src/Command/ChangeLogCommand.php

```diff
@@ -0,0 +1,214 @@
+    public function initialize(): void
+    {
+        $config = (array)Configure::read('ChangeLog');
+        $this->setConfig($config);
+    }
+
+    /**
+     * Read closed PRs from github and create changelog file.
+     *
+     * @param \Cake\Console\Arguments $args The command arguments.
+     * @param \Cake\Console\ConsoleIo $io The console io
+     * @return null|int The exit code or null for success
+     */
+    public function execute(Arguments $args, ConsoleIo $io): ?int
+    {
+        $from = $args->getArgument('from');
+        $io->out("Reading closed PRs from from $from");
+
+        $json = $this->fetchPrs($from);
+        $items = (array)Hash::get($json, 'items');
+        $io->out(sprintf('Loaded %d prs', count($items)));
+
+        $changeLog = $this->createChangeLog($items);
+        $this->saveChangeLog($changeLog, $args->getArgument('version'));
+
+        $io->out('Changelog created. Bye.');
+
+        return null;
+    }
+
+    /**
+     * Fetch pull requests from Github
+     *
+     * @param string $from From date.
+     * @return array
+     */
+    protected function fetchPrs(string $from): array
+    {
+        $client = new Client($this->getConfig('client'));
+        $query = [
+            'q' => sprintf('is:pr draft:false repo:bedita/bedita merged:>%s', $from),
+            'sort' => '-closed',
+            'per_page' => 100,
+        ];
+        $headers = ['Accept' => 'application/vnd.github.v3+json'];
+        $url = $this->getConfig('url');
+        $response = $client->get($url, $query, compact('headers'));
+
+        return (array)$response->getJson();
+    }
+
+    /**
+     * Create changelog array with classified pull requests
+     *
+     * @param array $items PR items
+     * @return array
+     */
+    protected function createChangeLog(array $items): array
+    {
+        $res = [];
+        foreach ($items as $item) {
+            $milestone = (string)Hash::get($item, 'milestone.title');
+            if (substr($milestone, 0, 1) !== '4') {
+                continue;
+            }
+            $labels = Hash::extract($item, 'labels.{n}.name');
+            $type = $this->classify($labels);
+            $res[$type][] = $item;
+        }
+
+        return $res;
+    }
+
+    /**
+     * Classify PR from its labels
+     *
+     * @param array $labels Labels array
+     * @return string
+     */
+    protected function classify(array $labels): string
+    {
+        foreach ((array)$this->getConfig('filter') as $name => $data) {
+            if (!empty(array_intersect($labels, $data))) {
+                return $name;
+            }
+        }
+
+        return 'other';
+    }
+
+    /**
+     * Save changelog to file
+     *
+     * @param array $changeLog Changelog items
+     * @param string $version Version released
+     * @return void
+     */
+    protected function saveChangeLog(array $changeLog, string $version): void
+    {
+        $out = sprintf("## Version %s - Cactus\n", $version);
```

Comment:
[minor] maybe it could be useful a parameter for the version alias as `Cactus` or a map to keep the couple `major version => alias`.

Reply:
Right... all releases are `Cactus` for now - we can change this once we decide for a new codename :wink: